subscriber.rb

```diff
@@ -1,21 +1,47 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
-
+require_relative 'q_it_errors'
 class Subscriber
 
   attr_reader :sqs, :sleep_period, :queue_url
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @queue_url = options[0].chomp
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
+    @queue_url = valid_sqs_url(options[0])
+    @sleep_period = validate_sleep_period(options[1])
+  end
+  
+  def valid_sqs_url(sqs_url)
+    valid_sqs_url_regex = /\A(https:\/\/sqs.)#{AWS["region"]}.amazonaws.com\/\d{12}\/[^.]+\z/
+    begin
+      raise QItNullSQSUrlError, "QItNullSQSUrlError :: #{self} AWS SQS URL cannot be nil." if sqs_url.nil?
+      raise QItInvalidSQSUrlError, "QItInvalidSQSUrlError :: #{self} Invalid SQS URL #{sqs_url}." unless sqs_url.match(valid_sqs_url_regex)
+      sqs_url
+    rescue QItNullSQSUrlError => e
```

Comment:
Same here - no `rescue` required.. break out. If we want to print the messages have a catch all, print the error and raise again.
Don't individually exit from all over the code - it will become difficult to track exit points.